app/AppComponents.scala

```diff
@@ -11,29 +12,22 @@ class AppComponents(context: Context) extends LoginControllerComponents(context,
 
   override val switches = new Switches(config, aws.s3Client)
 
-  private lazy val panDomainSettings: PanDomainAuthSettingsRefresher =
-    new PanDomainAuthSettingsRefresher(
-      domain = config.domain,
-      system = "login",
-      bucketName = config.pandaAuthBucket,
-      settingsFileKey = s"${config.domain}.settings",
-      s3Client = aws.s3Client
-    )
-
-  private lazy val desktopPanDomainSettings: PanDomainAuthSettingsRefresher = {
-    new PanDomainAuthSettingsRefresher(
-      domain = config.desktopDomain,
-      system = "login-desktop",
-      bucketName = config.pandaAuthBucket,
-      settingsFileKey = s"${config.desktopDomain}.settings",
-      s3Client = aws.s3Client
-    )
-  }
-
-  val loginPublicSettings = new PublicSettings(
-    settingsFileKey = s"${config.domain}.settings.public",
-    bucketName = config.pandaAuthBucket,
-    s3Client = aws.s3Client
+  private val s3BucketLoader: S3BucketLoader = forAwsSdkV1(aws.s3Client, "pan-domain-auth-settings")
+
+  private lazy val panDomainSettings: PanDomainAuthSettingsRefresher = PanDomainAuthSettingsRefresher(
+    domain = config.domain,
+    system = "login",
+    s3BucketLoader
+  )
+
+  private lazy val desktopPanDomainSettings: PanDomainAuthSettingsRefresher = PanDomainAuthSettingsRefresher(
+    domain = config.desktopDomain,
+    system = "login-desktop",
+    s3BucketLoader
+  )
+
+  val loginPublicSettings: PublicSettings = PublicSettings(
+    new Settings.Loader(s3BucketLoader, s"${config.domain}.settings.public")
```

Comment:
This PR:

* https://github.com/guardian/pan-domain-authentication/pull/151

...introduced the new `S3BucketLoader` abstraction, which simplifies constructing a `PanDomainAuthSettingsRefresher` and means that Panda is no longer _tied_ to AWS SDK v1 - an alternative AWS SDK v2 implementation of `S3BucketLoader` could be introduced.